api/relations/v0/relation_tuples.proto

```diff
@@ -0,0 +1,76 @@
+syntax = "proto3";
+
+package kessel.relations.v0;
+
+import "google/api/annotations.proto";
+import "relations/v0/common.proto";
+
+option go_package = "ciam-rebac/api/relations/v0;v0";
+option java_multiple_files = true;
+option java_package = "org.projectkessel.api.relations.v0";
+
+// KesselTupleServices manages the persisted _Tuples_ stored in the system..
+// 
+// A Tuple is an explicitly stated, persistent relation 
+// between a Resource and a Subject or Subject Set. 
+// It has the same _shape_ as a Relationship but is not the same thing as a Relationship.
+// 
+// A single Tuple may result in zero-to-many Relationships.
+service KesselTupleService {
+	rpc CreateTuples (CreateTuplesRequest) returns (CreateTuplesResponse) {
+		option (google.api.http) = {
+			post: "/v0/tuples"
+			body: "*"
+		};
+	};
+	rpc ReadTuples (ReadTuplesRequest) returns (ReadTuplesResponse) {
+		option (google.api.http) = {
+			get: "/v0/tuples"
+		};
+	};
+	rpc DeleteTuples (DeleteTuplesRequest) returns (DeleteTuplesResponse) {
+		option (google.api.http) = {
+			delete: "/v0/tuples"
+		};
+	};
+}
+
+message CreateTuplesRequest {
+	// Whether or not the request should ignore existing tuples (`true`),
+	// or if the request should fail if the same tuple already exists (`false`).
+	// 
+	// Defaults to `false`.
+	bool upsert = 1;
+	repeated Relationship tuples = 2;
+}
+message CreateTuplesResponse {}
+
+message ReadTuplesRequest {
+	RelationTupleFilter filter = 1;
+}
+message ReadTuplesResponse {
+	repeated Relationship tuples = 1;
```

Comment:
Should we this be streaming, or are we doing that as part of merging https://github.com/project-kessel/relations-api/pull/72 ?

Reply:
Yes and yes – I left that out for now, let's pick it up in #72

Reply:
Looks like you already updated that in #72 which is fine, so yeah let's just update it there